src/Lykke.blue.Service.ReferralLinks/Controllers/ReferralLinksController.cs

```diff
@@ -0,0 +1,480 @@
+                return await LogAndReturnBadRequest(request, ControllerContext, $"KYC needed for recipient client id {request.RecipientClientId} before claiming asset {refLink.Asset}");
+
+            var newRefLinkClaimRecipient = await CreateNewRefLinkClaim(refLink, request.RecipientClientId, true, request.IsNewClient);
+
+            var transactionRewardRecipient = await _exchangeService.TransferRewardCoins(refLink, request.IsNewClient, request.RecipientClientId, ControllerContext.GetControllerAndAction());
+
+            try
+            {
+                if (transactionRewardRecipient.IsOk())
+                {
+                    using (TransactionScope scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+                    {
+                        await SetRefLinkClaimTransactionId(transactionRewardRecipient, newRefLinkClaimRecipient);
+                        await UpdateRefLinkState(refLink, ReferralLinkState.Claimed);
+                        scope.Complete();
+                    }
+
+                    return Ok(new ClaimRefLinkResponse { TransactionRewardRecipient = transactionRewardRecipient.TransactionId, SenderOffchainTransferId = refLink.SenderOffchainTransferId });
+                }
+                return await LogAndReturnNotFound(request, ControllerContext, $"TransactionRewardRecipientError: Code: {transactionRewardRecipient.Code}, Message: {transactionRewardRecipient.Message}");
+            }
+            catch (TransactionAbortedException ex)
+            {
+                await LogError(new { Request = request, RefLink = refLink ?? new ReferralLink() }, ControllerContext, ex);
+                return NotFound(ex.Message);
+            }
+            catch (ApplicationException ex)
+            {
+                await LogError(new { Request = request, RefLink = refLink ?? new ReferralLink() }, ControllerContext, ex);
+                return NotFound(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                await LogError(new { Request = request, RefLink = refLink ?? new ReferralLink() }, ControllerContext, ex);
+                return NotFound(ex.Message);
+            }
+
+        }
+
+        /// <summary>
+        /// Claim invitation referral link.
+        /// </summary>
+        /// <param name="refLinkId"></param>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPut("invitationLinks/{refLinkId}/claim")]
+        [SwaggerOperation("ClaimInvitationLink")]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ClaimRefLinkResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> ClaimInvitationLink(string refLinkId, [FromBody] ClaimReferralLinkRequest request )
+        {
+            try
```

Comment:
The is too large 'try' block. Can be shortened.

Reply:
Shortened.